Documentation/proposals/20250122-blackbox-crd.md

```diff
@@ -0,0 +1,153 @@
+
+* **Other docs:**
+  * n/a
+
+This document proposes the creation of a Blackbox Operator and associated CRDs, similar to Prometheus-Operator, to manage blackbox-exporter instances and introduce a Module CRD for handling dynamic configuration of blackbox modules in a Kubernetes-native way.
+
+
+
+## Why
+The Probe CRD can define all the parameters that the ICMP Probe requires and is quite convenient for most use cases. But configuring the HTTP Probe using the Probe CRD is a lot more complex, requiring the user to define plenty of parameters. This makes the Probe CRD an insufficient solution and potentially error-prone for HTTP Probe use cases.
+
+The situation is very similar to Alertmanager/AlertmanagerConfig that prometheus-operator offers. We need a Kubernetes-native way to generate blackbox config dynamically by reading the Probe Module Config.
+
+
+## Pitfalls of the current solution
+
+The Probe CRD provides a partial solution for handling complex modules like HTTP and gRPC. However, it introduces a separate lifecycle for Blackbox management. Users must pre-define modules containing specific configurations (such as headers and request bodies) by manually updating and reloading the Blackbox configuration, before referencing these modules in their probes.
+
+## Goals
+
+- Provide a way for users to self-service adding module config for their Probe CRD
+- Consolidate the module configuration generation logic in a central point for other resources to use
+
+## Audience
+
+* Administrators providing Prometheus monitoring who want to enable self-service probe configuration for their users
+* Teams wanting to manage blackbox module configurations declaratively through Kubernetes CRDs
+* Users requiring a standardized, Kubernetes-native approach to monitor external targets
+
+## Non-Goals
+
+- This propose don't aim to replace the probe crd
+
+## How
+
+We propose creating two new Custom Resource Definitions (CRDs):
+- The BlackboxExporter CRD, which functions similarly to the Prometheus CRD, will manage the deployment configuration and lifecycle of blackbox-exporter instances
+- The BlackboxExporterModule CRD will define module specifications and automatically generate blackbox configurations, analogous to how Alertmanager works in Prometheus Operator
+
+## Alternatives
+Users may need to update or create a Probe CRD to monitor an HTTP endpoint with specific headers or body content. This often requires updating the Blackbox Exporter configuration, such as creating a new module or modifying an existing one. While users can manage this manually, it is a more cumbersome and error-prone process.
+
+## Action Plan
+
+1. Create the BlackboxExporter CRD which will manage blackbox-exporter instances as either a DaemonSet or Deployment
+2. Implement a configuration reloader sidecar container to automatically reload blackbox-exporter when configuration changes, similar to how Prometheus Operator handles configuration updates
```

Comment:
is there a need for another config reloader? I imagine that we can reuse the existing one.